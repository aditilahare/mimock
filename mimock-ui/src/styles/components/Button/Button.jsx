import React from 'react';
import PropTypes from 'prop-types';
import { ButtonContainer } from './Button.style';

function Button({ label, color, bgcolor }) {
	return (
		<ButtonContainer data-testid='button' color={color} bgcolor={bgcolor}>
			{label}
		</ButtonContainer>
	);
}

Button.propTypes = {
	label: PropTypes.string.isRequired,
	color: PropTypes.string,
	bgcolor: PropTypes.string,
};

export default Button;
