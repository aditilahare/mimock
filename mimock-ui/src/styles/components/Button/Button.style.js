import tw from "tailwind-styled-components";

export const ButtonContainer = tw.button`
  text-grey-200
  bg-blue-200
  p-4
  mx-auto
  my-auto
  shadow-md
  rounded-md
  hover:bg-blue-600
  cursor-pointer
  w-1/4
  my-4
  text-lg
  font-semibold
  ${(props) => Object.values(props)}
`;
