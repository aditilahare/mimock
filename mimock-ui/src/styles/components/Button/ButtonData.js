const ButtonData = {
	RED_BUTTON: {
		background: 'bg-red-300 dark:bg-red-500',
		color: 'text-red-800 dark:text-white',
	},
	DARK_RED_BUTTON: {
		background: 'bg-red-600',
		color: 'text-white',
	},
	GREEN_BUTTON: {
		background: 'bg-green-300 dark:bg-green-400',
		color: 'text-lime-700 dark:text-white',
	},
	DARK_GREEN_BUTTON: {
		background: 'bg-green-500',
		color: 'text-white',
	},
	INDIGO_BUTTON: {
		background: 'bg-blue-600',
		color: 'text-white',
	},
	TEAL_BUTTON: {
		background: 'bg-teal-400',
		color: 'text-white',
	},
	DEFAULT_BUTTON: {
		background: 'bg-white dark:bg-slate-300',
		color: 'text-black',
	},
};

const IconButtonData = {
	ADD_BUTTON: {
		background: 'bg-green-300 dark:bg-green-400',
		color: 'text-lime-700 dark:text-white',
	},
	DELETE_BUTTON: {
		background: 'bg-red-600',
		color: 'text-white',
	},
	CLEAR_BUTTON: {
		background: 'bg-red-300 dark:bg-red-500',
		color: 'text-red-800 dark:text-white',
	},
	COPY_BUTTON: {
		background: 'bg-blue-600',
		color: 'text-white',
	},
	FILTER_BUTTON: {
		background: 'bg-gray-300 dark:bg-slate-300',
		color: 'text-gray-600',
	},
	DEFAULT_BUTTON: {
		background: 'bg-white dark:bg-slate-300',
		color: 'text-black',
	},
	HOME_BUTTON: {
		background: 'bg-indigo-600 dark:bg-indigo-800',
		color: 'text-white',
	},
};

export { ButtonData, IconButtonData };
