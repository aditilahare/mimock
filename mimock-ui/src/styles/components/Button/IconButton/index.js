import IconButtonWithLabel from './IconButtonWithLabel/IconButtonWithLabel.jsx';
import IconButtonWithoutLabel from './IconButtonWithoutLabel/IconButtonWithoutLabel.jsx';

export { IconButtonWithLabel, IconButtonWithoutLabel };
