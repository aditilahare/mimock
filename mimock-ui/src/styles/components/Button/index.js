import Button from './DefaultButton/Button.jsx';
import CustomButton from './CustomButton/CustomButton.jsx';
import {
	IconButtonWithLabel,
	IconButtonWithoutLabel,
} from './IconButton/index';

export const ButtonVariants = {
	RedButton: 'RED_BUTTON',
	DarkRedButton: 'DARK_RED_BUTTON',
	GreenButton: 'GREEN_BUTTON',
	DarkGreenButton: 'DARK_GREEN_BUTTON',
	BlueButton: 'INDIGO_BUTTON',
	TealButton: 'TEAL_BUTTON',
	DefaultButton: 'DEFAULT_BUTTON',
};

export const IconButtonVariants = {
	ClearButton: 'CLEAR_BUTTON',
	DeleteButton: 'DELETE_BUTTON',
	AddButton: 'ADD_BUTTON',
	CopyButton: 'COPY_BUTTON',
	DefaultButton: 'DEFAULT_BUTTON',
};

export { Button, CustomButton, IconButtonWithLabel, IconButtonWithoutLabel };
