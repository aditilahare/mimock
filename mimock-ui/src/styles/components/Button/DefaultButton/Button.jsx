import React from 'react';
import PropTypes from 'prop-types';
import { ButtonContainer } from './Button.style';
import { ButtonData } from '../ButtonData';

function Button({ variant, label, width, onclickHandler, dataTestid }) {
	const { color, background } = getVariant(variant);

	return (
		<ButtonContainer
			data-testid={dataTestid}
			$color={color}
			$width={width}
			$background={background}
			onClick={onclickHandler}
		>
			{label}
		</ButtonContainer>
	);
}

function getVariant(variant) {
	return !variant ? ButtonData.DEFAULT_BUTTON : ButtonData[variant];
}

Button.propTypes = {
	label: PropTypes.string.isRequired,
	onclickHandler: PropTypes.func.isRequired,
	dataTestid: PropTypes.string,
	variant: PropTypes.string,
	width: PropTypes.string,
};

export default Button;
