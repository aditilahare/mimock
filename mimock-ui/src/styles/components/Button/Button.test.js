import React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
	let tree;

	it('should render Button', async () => {
		tree = await render(
			<Button label='Button' color='text-white' bgcolor='bg-black-100' />
		);

		const { getByTestId, container } = tree;

		expect(getByTestId('button')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render Button without style props', async () => {
		tree = await render(<Button label='Button' />);

		const { getByTestId, container } = tree;

		expect(getByTestId('button')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
