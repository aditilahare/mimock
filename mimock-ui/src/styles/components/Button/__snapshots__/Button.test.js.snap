// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button 1`] = `
<div>
  <button
    bgcolor="bg-black-100"
    class="text-grey-200 bg-blue-200 p-4 mx-auto my-auto shadow-md rounded-md hover:bg-blue-600 cursor-pointer w-1/4 my-4 text-lg font-semibold button text-white bg-black-100 Button"
    color="text-white"
    data-testid="button"
  >
    Button
  </button>
</div>
`;

exports[`Button should render Button without style props 1`] = `
<div>
  <button
    class="text-grey-200 bg-blue-200 p-4 mx-auto my-auto shadow-md rounded-md hover:bg-blue-600 cursor-pointer w-1/4 my-4 text-lg font-semibold button Button"
    data-testid="button"
  >
    Button
  </button>
</div>
`;
